### ISSUE TITLE
dom/wrap-form-element fails to remove missing props from state

### ISSUE THREAD

**Issue by thosmos** [fulcrologic/fulcro#357]
When a component adds and then removes a key to props, something like:
```clojure
     (ui-form-input (cond-> {:onChange (fn [evt]
                                         (m/set-string! this :user/name :event evt)
                                         (comp/transact! this [(fs/mark-complete! {:field :user/name})]))
                             :value name}
                      (fs/invalid-spec? user :user/name)
                      (assoc :error "User name cannot be blank")))
```
then the removed key does not also get removed from state.  

@codonnell has a fix here: https://github.com/codonnell/fulcro/commit/2542259d5fce6d6ca8221e6f869c8e5d9580d657
which moves the component's state under one child key called `data`.  This enables to component to add/remove keys, but may have consequences if existing users of this wrapper expect those values to be in the root of the state variable.  Is this possible?

**Comment by thosmos** [comment]
oops

**Comment by thosmos** [comment]
fixed in 9b2828131a0e4d5c098c0801542ea0b1d192cfd5
